CTriangle.cs

```diff
@@ -88,5 +88,39 @@ public void Print(string path)
             sw.WriteLine(this.ToString());
             sw.Close();
         }
+
+        public double Perimeter()
+        {
+            return ( firstSide.Length + secondSide.Length + thirdSide.Length );
+        }
+
+        public class ColoredTriangleComparer : IComparer<ColoredTriangle>
+        {
+            int IComparer<ColoredTriangle>.Compare(ColoredTriangle first, ColoredTriangle second)
+            {
+                return (first.Perimeter().CompareTo(second.Perimeter()));
+            }
+        }
+
+        public void ReadAndSort(string input_path, string output_path)
+        {
```

Comment:
Importand: you violate SOLID principles where S - single responsibility
https://en.wikipedia.org/wiki/SOLID

Reply:
Rename output_path -> outputPath

Reply:
Move function outside ColoredTriangle class and make it static